python/src/nanoarrow/c_array.py

```diff
@@ -298,7 +303,10 @@ def c_array_from_buffers(
     builder.resolve_null_count()
 
     # Validate + finish
-    return builder.finish(validation_level=validation_level)
+    if device is DEVICE_CPU:
+        return builder.finish(validation_level=validation_level)
+    else:
+        return builder.finish_device()
```

Comment:
This means it returns a ArrowArray vs ArrowDeviceArray depending on the device? (but you might also want a device array for cpu device?)